File: src/main/java/org/gangel/graphomance/arangodb/ArangoSessionProducer.java
package org.gangel.graphomance.arangodb;

import com.arangodb.ArangoDatabase;
import lombok.Builder;
import org.gangel.graphomance.Connection;
import org.gangel.graphomance.Session;
import org.gangel.graphomance.SessionProducer;

import java.util.Objects;

@Builder
public class ArangoSessionProducer implements SessionProducer {

    @Override
    public Session createSession(Connection connection) {
        ArangoConnection conn = Objects.requireNonNull((ArangoConnection) connection);

        ArangoDatabase db = conn.getDb()
                .db(conn.getConnectionSettings().getDbName());
        return ArangoSession.builder()
                .db(db)
                .connectionSettings(conn.getConnectionSettings())
                .build();
    }
}

File: src/main/kotlin/org/gangel/graphomance/arangodb/ArangoSessionProducer.kt
package org.gangel.graphomance.arangodb

import org.gangel.graphomance.Connection
import org.gangel.graphomance.Session
import org.gangel.graphomance.SessionProducer
import java.util.Objects

class ArangoSessionProducer : SessionProducer {
	override fun createSession(connection: Connection): Session {
		val conn: ArangoConnection = Objects.requireNonNull(connection as ArangoConnection)
		val db = conn.db
			.db(conn.connectionSettings
					.dbName)
		return ArangoSession(db = db, connectionSettings = conn.connectionSettings)
	}
}